File: app/src/main/java/com/example/firabasefirstexperience/LoginActivity.java
package com.example.firabasefirstexperience;

import android.app.ProgressDialog;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;


public class LoginActivity extends AppCompatActivity implements View.OnClickListener {


    private EditText eMail, password;
    private Button butLogIn;

    private FirebaseAuth auth;
    private FirebaseUser firebaseUser;

    private ProgressDialog progressDialog;

    @Override
    protected void onStart() {
        super.onStart();


    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        Toolbar toolbar = findViewById(R.id.tbForRegisterActivity);
        setSupportActionBar(toolbar);
        getSupportActionBar().setTitle("Sup Chat");
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);


        if(firebaseUser != null) {
            startActivity(new Intent(getApplicationContext(), MainActivity.class));
        }


        auth = FirebaseAuth.getInstance();



        progressDialog = new ProgressDialog(this);
        eMail = (EditText) findViewById(R.id.etEmail);
        password = (EditText) findViewById(R.id.etPassword);

        butLogIn = (Button) findViewById(R.id.butLogIn);

        butLogIn.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {

        int id = v.getId();

        switch (id) {
            case R.id.butLogIn:
                String textEmail = eMail.getText().toString();
                String textPassword = password.getText().toString();

                if (TextUtils.isEmpty(textEmail) || TextUtils.isEmpty(textPassword)) {
                    Toast.makeText(LoginActivity.this, "All fields must be filled", Toast.LENGTH_SHORT).show();
                } else {
                    auth.signInWithEmailAndPassword(textEmail, textPassword)
                            .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                                @Override
                                public void onComplete(@NonNull Task<AuthResult> task) {
                                    if (task.isSuccessful()) {
                                        Intent intent = new Intent(getApplicationContext(), MainActivity.class);
                                        startActivity(intent);
                                        Toast.makeText(LoginActivity.this, "Welcome to your Account!", Toast.LENGTH_SHORT).show();

                                        finish();
                                    } else {
                                        Toast.makeText(LoginActivity.this, "Authentication failed", Toast.LENGTH_SHORT).show();
                                    }
                                }
                            });
                }

        }
    }
}

File: app/src/main/java/com/example/firabasefirstexperience/kotlin/presentation/login/ui/LoginActivity.kt
package com.example.firabasefirstexperience.kotlin.presentation.login.ui

import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.widget.Toast
import com.arellomobile.mvp.MvpAppCompatActivity
import com.arellomobile.mvp.presenter.InjectPresenter
import com.example.firabasefirstexperience.R
import com.example.firabasefirstexperience.kotlin.presentation.login.mvp.LoginActivityPresenter
import com.example.firabasefirstexperience.kotlin.presentation.login.mvp.LoginActivityView
import com.example.firabasefirstexperience.kotlin.presentation.main.ui.MainActivity
import com.example.firabasefirstexperience.kotlin.presentation.resetpassword.ui.ResetPassActivity
import kotlinx.android.synthetic.main.activity_login.*
import kotlinx.android.synthetic.main.bar_layout.*

class LoginActivity : MvpAppCompatActivity(), LoginActivityView {

    @InjectPresenter
    lateinit var presenter: LoginActivityPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        initViews()
    }

    private fun initViews() {

        setSupportActionBar(tbForRegisterActivity)
        supportActionBar?.title = "Sup Chat"
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        butLogIn.setOnClickListener {
            val textEmail = etEmail.text.toString()
            val textPassword = etPassword.text.toString()
            if (TextUtils.isEmpty(textEmail) || TextUtils.isEmpty(textPassword)) {
                presenter.onEmptyError()
            } else {
                presenter.userLog(textEmail, textPassword)
            }
        }

        tvResetPassword.setOnClickListener {
            presenter.userResetPass()
        }
    }

    override fun showEmptyError() {
        Toast.makeText(applicationContext, "All fields must be filled!", Toast.LENGTH_SHORT).show()
    }

    override fun showSuccess() {
        Toast.makeText(applicationContext, "Welcome to your Account", Toast.LENGTH_SHORT).show()
    }

    override fun showAuthError() {
        Toast.makeText(applicationContext, "Authentication failed. Wrong password or email", Toast.LENGTH_LONG).show()
    }

    override fun openResetPass() {
        startActivity(Intent(this@LoginActivity, ResetPassActivity::class.java))
    }

    override fun openMain() {
        startActivity(Intent(this@LoginActivity, MainActivity::class.java))
    }
}